src/pages/settings/company/create/CompanyCreate.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Button } from '@invoiceninja/forms';
+import { AxiosError } from 'axios';
+import { AuthenticationTypes } from 'common/dtos/authentication';
+import { endpoint } from 'common/helpers';
+import { request } from 'common/helpers/request';
+import { route } from 'common/helpers/route';
+import { toast } from 'common/helpers/toast/toast';
+import { updateCompanyUsers } from 'common/stores/slices/company-users';
+import { authenticate } from 'common/stores/slices/user';
+import { Modal } from 'components/Modal';
+import { useState, SetStateAction, Dispatch } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useQueryClient } from 'react-query';
+import { useDispatch } from 'react-redux';
+
+interface Props {
+  isModalOpen: boolean;
+  setIsModalOpen: Dispatch<SetStateAction<boolean>>;
+}
+
+export function CompanyCreate(props: Props) {
+  const [t] = useTranslation();
+
+  const dispatch = useDispatch();
+
+  const queryClient = useQueryClient();
+
+  const [isFormBusy, setIsFormBusy] = useState<boolean>(false);
+
+  const switchCompany = (
+    index: number,
+    passedUser: Record<string, unknown>,
+    passedToken: string
+  ) => {
+    dispatch(
+      authenticate({
+        type: AuthenticationTypes.TOKEN,
+        user: passedUser,
+        token: passedToken,
+      })
+    );
+
+    localStorage.setItem('X-CURRENT-INDEX', index.toString());
+
+    sessionStorage.setItem('COMPANY-EDIT-OPENED', 'false');
+
+    queryClient.invalidateQueries();
+
+    window.location.href = route('/');
+  };
+
+  const handleSave = async () => {
+    if (!isFormBusy) {
+      toast.processing();
+
+      setIsFormBusy(true);
+
+      try {
+        await request('POST', endpoint('/api/v1/companies'));
+
+        const response = await request('POST', endpoint('/api/v1/refresh'));
+
+        const companyUsers = response.data.data;
+
+        const createdCompanyIndex = companyUsers.length - 1;
+
+        const companyUser = companyUsers[createdCompanyIndex];
+
+        dispatch(updateCompanyUsers(companyUsers));
+
+        toast.success('created_new_company');
+
+        props.setIsModalOpen(false);
+
+        switchCompany(
+          createdCompanyIndex,
+          companyUser.user,
+          companyUser.token.token
+        );
+      } catch (error) {
+        const axiosError = error as AxiosError;
```

Comment:
The more I look into this code, the more I don't like this try-catch block for single responses. If you are chaining them, for sure, for single, please use promise-based style because you can pass the type and generic interface to the .then or .catch blocks themselves. This way you are essentially just "lying" to the compiler about the type, and if the underline type of `AxiosError` changes, we wouldn't know because you are the one explicitly casting it.

Reply:
changed